### PR TITLE
KS onboarding to UP

### PR DESCRIPTION
**Pull request by oslynn**
This PR is migrating KS from Covid pipeline to UP.  They only need covid and HIV result to be reported.

Test Steps:
1.) clone the PR
2.) gradlew clean package
3.) ./gradlew testIntegration
4.) It should SUCCESS

## Changes
- *Include a comprehensive list of changes in this PR*
- *(For web UI changes) Include screenshots/video of changes*

### Testing
- [x] Tested locally?
- [x] Ran `./prime test` or `./gradlew testSmoke` against local Docker ReportStream container?
- [ ] (For Changes to /frontend-react/...) Ran `npm run lint:write`? 
- [ ] Added tests?

### Process
- [ ] Are there licensing issues with any new dependencies introduced?
- [ ] Includes a summary of what a code reviewer should test/verify?
- [ ] Updated the release notes?
- [ ] Database changes are submitted as a separate PR?
- [ ] DevOps team has been notified if PR requires ops support?

## Linked Issues
- Fixes #issue
- https://app.zenhub.com/workspaces/engagement-6166edbd409257001e09f1b2/issues/gh/cdcgov/prime-reportstream/14001

## To Be Done
*Create GitHub issues to track the work remaining, if any*
- #issue

## Specific Security-related subjects a reviewer should pay specific attention to
- Does this PR introduce new endpoints?
    - new endpoint A
    - new endpoint B
- Does this PR include changes in authentication and/or authorization of existing endpoints?
- Does this change introduce new dependencies that need vetting?
- Does this change require changes to our infrastructure?
- Does logging contain sensitive data?
- Does this PR include or remove any sensitive information itself?

If you answered '_yes_' to any of the questions above, conduct a detailed Review that addresses at least:

- What are the potential security threats and mitigations? Please list the _STRIDE_ threats and how they are mitigated
    - **S**poofing (faking authenticity)
        - Threat _T_, which could be achieved by _A_, is mitigated by _M_
    - **T**ampering (influence or sabotage the integrity of information, data, or system)
    - **R**epudiation (the ability to dispute the origin or originator of an action)
    - **I**nformation disclosure (data made available to entities who should not have it)
    - **D**enial of service (make a resource unavailable)
    - **E**levation of Privilege (reduce restrictions that apply or gain privileges one should not have)
- Have you ensured logging does not contain sensitive data?
- Have you received any additional approvals needed for this change?

### CODE REVIEW

File: prime-router/src/main/resources/metadata/hl7_mapping/receivers/STLTs/KS/KS-receiver-transform.yml [hunk 1]
```diff
@@ -0,0 +1,59 @@
...
+elements:
+
+  - name: xtn-use-code-from-system
+    condition: '%resource.system = "email"'
+    value: [ '"NET"' ]
+    hl7Spec: [ '%{hl7TelecomPath}-2' ]
+
+  - name: ks-message-encoding-characters
+    condition: true
+    value: [ '"^~\\&"' ]
```
Comment (victor-chaparro) [inline prime-router/src/main/resources/metadata/hl7_mapping/receivers/STLTs/KS/KS-receiver-transform.yml:14]: Why is this needed? All message should have this encoding characters by default. We only need to override it if it's something different.

File: prime-router/src/main/resources/metadata/hl7_mapping/receivers/STLTs/KS/KS-receiver-transform.yml [hunk 2]
```diff
@@ -0,0 +1,59 @@
...
+    hl7Spec: [ 'MSH-6-3' ]
+
+  - name: ks-patient-race-coding-system
+    resource: 'Bundle.entry.resource.ofType(Patient).extension("http://ibm.com/fhir/cdm/StructureDefinition/local-race-cd").value.coding'
+    condition: '%resource.code.exists()'
+    value: [ '%resource.system.getCodingSystemMapping()' ]
+    hl7Spec: [ '/PATIENT_RESULT/PATIENT/PID-10-3' ]
+
+  - name: observation-result-with-aoe
+    # Grab only the AOE observations from ServiceRequest.supportingInfo NOT associated with a specimen
```
Comment (victor-chaparro) [inline prime-router/src/main/resources/metadata/hl7_mapping/receivers/STLTs/KS/KS-receiver-transform.yml:54]: We are not suppressing AOEs in the covid pipeline. Why are we suppressing them in the UP?

File: prime-router/src/main/resources/metadata/hl7_mapping/receivers/STLTs/KS/KS-receiver-transform.yml (hunk 1 above)
Comment (oslynn) [reply prime-router/src/main/resources/metadata/hl7_mapping/receivers/STLTs/KS/KS-receiver-transform.yml:14]: Ok, it will be removed.

File: prime-router/src/main/resources/metadata/hl7_mapping/receivers/STLTs/KS/KS-receiver-transform.yml (hunk 2 above)
Comment (oslynn) [reply prime-router/src/main/resources/metadata/hl7_mapping/receivers/STLTs/KS/KS-receiver-transform.yml:54]: KS doesn't need AOE.

File: prime-router/src/main/resources/metadata/hl7_mapping/receivers/STLTs/KS/ks-doh.yml [hunk 3]
```diff
@@ -0,0 +1,100 @@
...
+        - "(Bundle.entry.resource.ofType(Patient).address.line.exists() or Bundle.entry.resource.ofType(Patient).address.postalCode.exists() or Bundle.entry.resource.ofType(Patient).telecom.exists())"
+        - "Bundle.entry.resource.ofType(MessageHeader).id.exists()"
+        - "Bundle.entry.resource.ofType(Patient).birthDate.exists()"
+        - "Bundle.entry.resource.ofType(Patient).name.family.exists()"
+        - "Bundle.entry.resource.ofType(Patient).name.given.count() > 0"
+        - "Bundle.entry.resource.ofType(Specimen).type.exists()"
+      processingModeFilter: []
+      reverseTheQualityFilter: false
+      conditionFilter:
+        - "(%resource.interpretation.coding.code = 'A' and %resource.code.coding.extension('https://reportstream.cdc.gov/fhir/StructureDefinition/condition-code').value.where(code in ('840539006')).exists()) or  (%resource.code.coding.extension('https://reportstream.cdc.gov/fhir/StructureDefinition/condition-code').value.where(code in ('6142004' | '541131000124102')).exists())"
```
Comment (oslynn) [inline prime-router/src/main/resources/metadata/hl7_mapping/receivers/STLTs/KS/ks-doh.yml:77]: Need to remove %resource.interpretation.coding.code = 'A' and 
KS wants all +/- HIV

**Review by victor-chaparro — APPROVED** [review]
